apps/frontend/src/modules/surveys/components/studio/create/editor-panel/components/EditorTextInput.tsx

```diff
@@ -21,89 +25,137 @@ export const EditorTextInput = ({
+    const [textContent, setTextContent] = useState(decode(defaultValue ?? ""));
+
+    const [displayFallbackField, setDisplayFallbackField] = useState(false);
+
+    const modules: StringMap = useMemo(() => {
+        return {
+            toolbar: false,
+            syntax: false,
+            mention: {
+                allowedChars: /^[A-Za-z\sÅÄÖåäö]*$/,
+                allowInlineMentionChar: true,
+                isolateCharacter: true,
+                mentionDenotationChars: ["@"],
+                showDenotationChar: false,
+                spaceAfterInsert: true,
+                positioningStrategy: "absolute",
+                source: function (searchTerm, renderList, mentionChar) {
+                    if (searchTerm.startsWith(" ")) {
+                        renderList([]);
+                        return;
+                    }
+                    renderList(options);
+                },
+                onSelect: function (item, insertItem) {
+                    const newItem = item;
+                    const details = newItem.id.split(" ");
+                    insertItem({ ...newItem, value: item.value, id: details[0], type: details[1] });
+                },
+            },
+        };
+    }, []);
+
+    function resolveQuestionIndex(questionId: string): string {
+        const option = options?.find((o) => {
+            const optionId = o.id.split(" ")[0];
+            return questionId === optionId;
+        });
+
+        return option?.value as string;
+    }
+
+    function encode(textContent: string): string {
+        // Replace span tags with placeholders
+        const encodedText = textContent.replace(
+            /<span class="mention"([^>]*)data-id="([^"]*)"([^>]*)data-type="([^"]*)"([^>]*)>(.*?)<\/span>/g,
+            (_, __, dataId, ___, dataType) => {
+                if (dataId === "attribute") {
+                    return `{{${dataType}}}`;
+                }
+                return `{{${dataType}:${dataId}}}`;
+            },
+        );
+
+        return encodedText.split("</span>").join("");
+    }
+
+    function decode(encodedText: string): string {
+        // Replace placeholders with actual HTML tags
+        const decodedText = encodedText
+            .replace(/{{answer:([^}]+)}}/g, (match, id) => {
+                return `<span class="mention" data-index="4" data-denotation-char="" data-value="${resolveQuestionIndex(id)}" data-id="${id}" data-type="answer">﻿<span contenteditable="false">${resolveQuestionIndex(id)}</span>﻿</span>`;
+            })
+            .replace(/{{attribute.([^}]+)}}/g, (_, attr) => {
+                return `<span class="mention" data-index="4" data-denotation-char="" data-value="${attr}" data-id="attribute" data-type="attribute.${attr}">﻿<span contenteditable="false">${attr}</span>﻿</span>`;
+            });
+
+        return decodedText + " ";
+    }
+
+    const handleMentionClicked = (event: Event) => {
+        setDisplayFallbackField(true);
```

Comment:
@afolasope Why not use `setDisplayFallbackField` directly. You don't need to declare a new function.